2216/2/top_words_analyzer.rb

```diff
@@ -0,0 +1,65 @@
+class TopWordsAnalyzer
+  def initialize(battles, top_number)
+    @battles = battles
+    @output_num = top_number
+    @words = divide_into_words
+  end
+
+  def analyze_top
+    counts = {}
+    make_counts(counts)
+    counts = counts.sort_by { |_key, value| value }.reverse
+    output_words(counts)
+  end
+
+  def self.print_top_words(word_with_count)
+    puts "#{word_with_count[0]} - #{word_with_count[1]} раз"
+  end
+
+  private
+
+  def make_counts(counts)
```

Comment:
Why do you want to pass counts as the argument?
You can just return what you need.

Reply:
Moreover, the method name is not self-explanatory. It's not quite clear what kind of counts you want to create here.